### DISCUSSION TITLE
Image dependency dist path

### DISCUSSION THREAD

**Discussion by studiosciences** [parcel-bundler/parcel#9249] [Q&A] [answered]
I'm working an MDX2 transformer that automatically adds image dependencies, which is working, but I haven't figured out the best option for getting the final image URL. Ideally, I would let the plugins handle the file import as normal.

Based on other transformers, it appears that `asset.addURLDependency(url)` should return the path, but that's not what I get. Neither have I figured out how to define my own final path when calling that function.

```
const src = asset.addURLDependency('./screenshot.png');
 // ACTUAL: src = d8ebae247c1d5c48
 // WANT: src = '/screenshot.22bf891f.png' (dist output )
```

I'd be happy to pass in my own final path, but I haven't figured out how to do that. The documentation for `loc`, `fs` and `filename` doesn't do a great job of explaining how they are used in this case.

`needsStablePath` is ok, but not ideal and I still need to calculate the path.

**Comment by mischnic** [comment, marked as answer]
The final path isn't known yet when the transformer runs, that's why that id is returned. 

The intended way is: just include that id, and then `replaceURLReferences` in the packager (e.g. the html or the raw-url packager) will replace it at the end once the bundle name is known.

If that doesn't work for you: why do you need the final url in the transformer?

**Reply by studiosciences** [reply]
Of course, I immediately find this after asking...
https://github.com/parcel-bundler/parcel/discussions/5393

`parcel/packager-raw-url` should solve the issue, thx.

**Reply by studiosciences** [reply]
I found the following comment in the code, but I'm not seeing this behavior in the code.
> For library builds, we need to replace URL references with their final resolved paths.
 For non-library builds, this is handled in the JS runtime.

Should I be inserting the ID into the JS differently? Here is the compiled code...
```
                children: [
                    (0, _jsxDevRuntime.jsxDEV)(_components.img, {
                        src: "c26020bc230257cf", // SHOULD BE THE CORRECT URL. 
                        alt: "Step 1"
                    }, undefined, false, {
```

Flow within my Transformer:
- Read 'mdx' file as `code`
- Extract image urls from `code`.
- Add urls as dependencies.
- Replace urls in `code` with the ids.
- Run the updated `code` through mdx compiler

**Reply by studiosciences** [reply]
Nevermind. Found an MDX plugin that solves all my problems.
